src/badge.jsx

```diff
@@ -0,0 +1,119 @@
+const React = require('react');
+const Typography = require('./styles/typography');
+const DefaultRawTheme = require('./styles/raw-themes/light-raw-theme');
+const ThemeManager = require('./styles/theme-manager');
+
+// Badge
+export default React.createClass({
+  displayName: 'Badge',
+  contextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+  //for passing default theme context to children
+  childContextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+  getChildContext () {
+    return {
+      muiTheme: this.state.muiTheme,
+    };
+  },
+  propTypes: {
+    className: React.PropTypes.string,
+    primary: React.PropTypes.bool,
+    secondary: React.PropTypes.bool,
+    primaryText: React.PropTypes.string.isRequired,
+    labelColor: React.PropTypes.string,
+    backgroundColor: React.PropTypes.string,
+    style: React.PropTypes.object,
+  },
+  getInitialState() {
+    return {
+      muiTheme: this.context.muiTheme ? this.context.muiTheme : ThemeManager.getMuiTheme(DefaultRawTheme),
+    };
+  },
+  getDefaultProps() {
+    return {
+      className: '',
+      primary: false,
+      secondary: false,
+      primaryText: '',
+      labelColor: null,
+      backgroundColor: null,
+      style: null,
+    };
+  },
+  componentWillReceiveProps(nextProps, nextContext) {
+    let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
+    this.setState({
+      muiTheme: newMuiTheme,
+    });
+  },
+  getTheme() {
+    return this.state.muiTheme.badge;
+  },
+  getStyles() {
+    const backgroundColor = this.props.backgroundColor
+      ? this.props.backgroundColor
+      : this.props.primary
+        ? this.getTheme().primaryColor
```

Comment:
getTheme() will only be invoked once, so it might be more efficient not to store an unnecessary variable, but if you want to do it that way for legibility it's fine with me.